state/unit.go

```diff
@@ -380,22 +386,72 @@ func (op *UpdateUnitOperation) Build(attempt int) ([]txn.Op, error) {
 			Updated:    now.UnixNano(),
 		}
 		op.setStatusDocs[key] = doc
-		statusOps, err := statusSetOps(op.unit.st.db(), doc, key)
+		// It's possible we're getting a first status update (i.e. cloud container)
+		_, err = getStatus(op.unit.st.db(), key, "")
 		if err != nil {
-			return errors.Trace(err)
+			if !errors.IsNotFound(err) {
+				return errors.Trace(err)
+			}
+			statusOps := createStatusOp(op.unit.st, key, doc)
+			ops = append(ops, statusOps)
+		} else {
+			statusOps, err := statusSetOps(op.unit.st.db(), doc, key)
+			if err != nil {
+				return errors.Trace(err)
+			}
+			ops = append(ops, statusOps...)
 		}
-		ops = append(ops, statusOps...)
 		return nil
 	}
 	if op.props.AgentStatus != nil {
 		if err := updateStatus(op.unit.globalAgentKey(), op.props.AgentStatus); err != nil {
 			return nil, errors.Trace(err)
 		}
 	}
+
+	var unitStatus, cloudContainerStatus status.StatusInfo
 	if op.props.UnitStatus != nil {
 		if err := updateStatus(op.unit.globalKey(), op.props.UnitStatus); err != nil {
 			return nil, errors.Trace(err)
 		}
+		unitStatus = *op.props.UnitStatus
+	}
+	if op.props.CloudContainerStatus != nil {
+		if err := updateStatus(op.unit.globalCloudContainerKey(), op.props.CloudContainerStatus); err != nil {
+			return nil, errors.Trace(err)
+		}
+		cloudContainerStatus = *op.props.CloudContainerStatus
+	}
+	if unitStatus.Status != "" || cloudContainerStatus.Status != "" {
```

Comment:
I think with the rework, the only unit status we can get here is Terminated. Right?
Which means we don't need to bother reloading the container status. Terminated wins.